dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/utils/DagHelper.java

```diff
@@ -384,13 +384,40 @@ public static List<Long> parseConditionTask(Long nodeCode,
         // the skipNodeList maybe null if no next task
         skipNodeList = Optional.ofNullable(skipNodeList).orElse(new ArrayList<>());
         for (Long failedNode : skipNodeList) {
-            setTaskNodeSkip(failedNode, dag, completeTaskList, skipTaskNodeList);
+            // return nodes set to skip in setTaskNodeSkip method
+            Map<Long, TaskNode> skipNodesAfterFailedNode =
+                    setTaskNodeSkip(failedNode, dag, completeTaskList, skipTaskNodeList);
+            parseConditionNodeAfterSkippedNode(skipTaskNodeList, dag, conditionTaskList, skipNodesAfterFailedNode);
         }
         // the conditionTaskList maybe null if no next task
         conditionTaskList = Optional.ofNullable(conditionTaskList).orElse(new ArrayList<>());
         return conditionTaskList;
     }
 
+    /**
+     * In some special cases, such as the condition task is set to "OR", the condition task after skipped node needs to be submitted to execution
+     *
+     * @param skipTaskNodeList
+     * @param dag
+     * @param conditionTaskList
+     * @param skipTaskAfterNode
+     */
+    private static void parseConditionNodeAfterSkippedNode(Map<Long, TaskNode> skipTaskNodeList,
+                                                           DAG<Long, TaskNode, TaskNodeRelation> dag,
+                                                           List<Long> conditionTaskList,
+                                                           Map<Long, TaskNode> skipTaskAfterNode) {
+
+        for (Long taskCode : skipTaskAfterNode.keySet()) {
+            Set<Long> subsequentNodes = dag.getSubsequentNodes(taskCode);
```

Comment:
I am puzzled by this, the ```skipTaskAfterNode ```parameters are all nodes not included in the ```dag```, so can you still get them through the ```dag```?

Reply:
I know what's going on, ignore the above question